src/openrct2/world/Sprite.cpp

```diff
@@ -728,6 +755,10 @@ void SpriteBase::MoveTo(const CoordsXYZ& newLocation)
     {
         loc.x = LOCATION_NULL;
     }
+    else
```

Comment:
Why the else? If we are picking up a peep we move it to a LOCATION_NULL we would still want to invalidate its origin.

Reply:
Oh good catch, I misread the code a bit, I thought this is checking the old position but that makes no sense. I'll change it to what I thought: Invalidate if previous pos is valid and Invalidate when the new location is valid.